compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -46,7 +47,10 @@ impl MetadataLoader for DefaultMetadataLoader {
                 let entry = entry_result
                     .map_err(|e| format!("failed to parse rlib '{}': {}", path.display(), e))?;
                 if entry.name() == METADATA_FILENAME.as_bytes() {
-                    return Ok(entry.data());
+                    let data = entry
+                        .data(data)
+                        .map_err(|e| format!("failed to parse rlib '{}': {}", path.display(), e))?;
+                    return search_for_metadata(path, data);
```

Comment:
Isn't code to handle raw metadata files missing?

Reply:
It happens just below in `search_for_metdata` which returns the original slice if `object::File` doesn't parse.